core-bundle/src/Controller/ContentElement/DownloadsController.php

```diff
@@ -222,14 +222,22 @@ private function getPreviews(FilesystemItem $filesystemItem, ContentModel $model
         }
     }
 
-    private function handleDownload(Request $request): void
+    private function handleDownload(Request $request, ContentModel $model): void
     {
         $response = $this->fileDownloadHelper->handle(
             $request,
             $this->filesStorage,
-            function (FilesystemItem $item, array $context): Response|null {
+            function (FilesystemItem $item, array $context) use ($model): Response|null {
+                $id = $context['id'] ?? null;
+
+                // Do not handle downloads from other DownloadController
+                // elements on the same page (see #5568)
+                if ($id !== $model->id) {
+                    return new Response('', Response::HTTP_NO_CONTENT);
+                }
+
                 if (
-                    null === ($model = $this->getContaoAdapter(ContentModel::class)->findById($context['id'] ?? null)) ||
+                    null === ($model = $this->getContaoAdapter(ContentModel::class)->findById($id)) ||
```

Comment:
We should rename this parameter to `$contentModel` or `$contextModel` or something else that does not interfere with the given `use ($model)`. WDYT?

Reply:
Agree, that would be cleaner if the variable was not redeclared. In fact as the retrieved model is identical to the now passed argument, the logic can be further simplified. 

Changed in a50d049f9eac6f4d45892306aac888b88d7afe33.